src/test/java/org/matsim/prepare/longDistanceFreightGER/tripGeneration/GenerateFreightTripTest.java

```diff
@@ -0,0 +1,72 @@
+package org.matsim.prepare.longDistanceFreightGER.tripGeneration;
+
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.RegisterExtension;
+import org.matsim.api.core.v01.network.Network;
+import org.matsim.api.core.v01.population.Person;
+import org.matsim.api.core.v01.population.Population;
+import org.matsim.api.core.v01.population.PopulationWriter;
+import org.matsim.core.config.ConfigUtils;
+import org.matsim.core.network.NetworkUtils;
+import org.matsim.core.population.PopulationUtils;
+import org.matsim.testcases.MatsimTestUtils;
+
+import java.io.IOException;
+import java.util.List;
+
+import static org.matsim.prepare.longDistanceFreightGER.tripGeneration.GenerateFreightPlans.createOutput_tripOD_relations;
+
+public class GenerateFreightTripTest {
+
+	private static final Logger log = LogManager.getLogger(GenerateFreightTripTest.class);
+
+	@RegisterExtension
+	public MatsimTestUtils utils = new MatsimTestUtils() ;
+
+	// TODO the test fails... But the main class with the same input arguments can run without problem... I don't know why
+	@Test
+	public final void test() {
+		String [] args = {
+			"--output", utils.getOutputDirectory(),
+			"--sample", "0.001",
+			"--land-use-filter"
+		};
+
+		GenerateFreightPlans.main(args);
+	}
+
+	@Test
```

Comment:
Hi, what does this test is testing?
I would assume, that you are not doing all loggings, but instead /additionally check for some results. see further comments.

At the end (or as another test), you could also do a check 
1) if the populationsFile is identical to an expected one (there is a method for this),
2) if the freight_trips_data.tsv is identical (there is a method for doing a line-by-line compare)

Thanks a lot